src/redux/reducers/header.reducer.js

```diff
@@ -0,0 +1,12 @@
+import { HEADER_ACTIONS } from '../actions/header.actions'
+
+export default ( state = '', action) => {
+	switch (action.type) {
+		case HEADER_ACTIONS.UPDATE_SEARCH_TEXT:
+			return 	{...state, searchText: action.payload}
+		case HEADER_ACTIONS.GET_ALCOHOL_LIST:
+			return {...state, alcohol: action.payload}
+		default:
+		 return state
+	}
+}
```

Comment:
Hi... the search text is updating on state just fine. I would say that we don't also need to set 'alcohol' on state, i.e., we don't need this part of the reducer:
`case HEADER_ACTIONS.GET_ALCOHOL_LIST:
       return {...state, alcohol: action.payload}`

I think that the action GET_ALCOHOL_LIST is what the epic will listen for. Then the epic will make the API call and return a response with one of two actions - one action for success, one action for failure. Those epics will then go to a reducer that will set search results on state. Does that make sense?